apps/mobile/src/screens/ProfilePicturePickerScreen/ProfilePicturePickerSingularAsset.tsx

```diff
@@ -37,44 +38,30 @@ export function ProfilePicturePickerSingularAsset({
     tokenRef
   );
 
-  const [setProfileImage, isSettingProfileImage] =
-    usePromisifiedMutation<ProfilePicturePickerSingularAssetMutation>(graphql`
-      mutation ProfilePicturePickerSingularAssetMutation($input: SetProfileImageInput!) {
-        setProfileImage(input: $input) {
-          ... on SetProfileImagePayload {
-            viewer {
-              user {
-                ...ProfilePictureFragment
-              }
-            }
-          }
-        }
-      }
-    `);
+  const route = useRoute<RouteProp<RootStackNavigatorParamList, 'Post'>>();
+  const currentScreen = route.params.page;
+
+  const navigation = useNavigation<MainTabStackNavigatorProp>();
+
+  const { setProfileImage, isSettingProfileImage } = useProfilePicture();
 
-  const reportError = useReportError();
   const tokenUrl = getVideoOrImageUrlForNftPreview({ tokenRef: token })?.urls.large;
 
   const [, setError] = useState<string | null>(null);
 
   const handlePress = useCallback(() => {
     setError(null);
 
-    setProfileImage({
-      variables: {
-        input: {
-          tokenId: token.dbid,
-        },
-      },
-    })
-      .then(() => {
+    if (currentScreen === 'ProfilePicture') {
+      setProfileImage(token.dbid).then(() => {
         onProfilePictureChange();
-      })
-      .catch((error) => {
-        setError(error);
-        reportError(error);
       });
-  }, [onProfilePictureChange, reportError, setProfileImage, token.dbid]);
+    } else {
+      navigation.navigate('PostComposer', {
```

Comment:
since we're using this `ProfilePicturePickerSingularAsset.tsx` for the Posts flow we should also rename this to be more generic like "NftSelectorSingularAsset`, along with all other references to profile pics in this file

Reply:
below on lines 71 and 72 there are some analytics events specific to Profile Pictures which shouldn't be sent if this component is being used for the Posts flow

Reply:
I did see that you isolated the pfp setting function and moved it out of this file though 🔥🔥 so renaming the other stuff will just help complete the separation